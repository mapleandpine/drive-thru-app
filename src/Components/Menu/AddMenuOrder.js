import React, { Component } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';

class AddMenuOrder extends Component {

  constructor(props) {
    super(props);
    this.state = {
      newMenuItem:{}
    }
  }

  handleSubmit(e) {
    if(this.refs.title.value === "") {
      alert('Title is required');
    } else {
      this.setState({newMenuItem:{
        id: uuid.v4(),
        title: this.refs.title.value,
        category: this.refs.category.value
      }}, function() {
        this.props.addMenuItem(this.state.newMenuItem);
      })
    }
    e.preventDefault();
  }

  render() {
    let categoryOptions = this.props.categories.map(category => {
      return <option key={category} value={category}>{category}</option>
    })
    return (
      <div className="AddMenuItem">
        <h3>Add Order</h3>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <div>
            <label>Title</label><br />
            <input type="text" ref="title" />
          </div>
          <br />
          <div>
            <label>Payment</label><br />
            <select ref="category">
              {categoryOptions}
            </select>
          </div>
          <br />
          <input type="submit" value="Submit"/>
          <br />
        </form>
      </div>
    );
  }
}

AddMenuOrder.propTypes = {
  categories: PropTypes.array,
  addMenuOrder: PropTypes.func
}

export default AddMenuOrder;
