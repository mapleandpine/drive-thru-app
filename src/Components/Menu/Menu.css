/* MENU */

.Menu {
  background-color: #555;
  border-radius: 0 0 12px 0;
  grid-column: 1;
  grid-row: 1;
  padding: 6px;
}

.Menu .heading {
  text-transform: uppercase;
}

.MenuList {
  display: flex;
  flex-wrap: wrap;
}

.MenuItem {
  padding: 5px;
  width: 25%;
}

.MenuItem figure {
  background-color: #e5e5e5;
  background-repeat: no-repeat;
  background-size: 400%;
  border-radius: 8px;
  box-shadow: 0 0 40px rgba(255,255,255,.4) inset;
  color: #353535;
  cursor: pointer;
  height: 140px;
  justify-content: center;
  padding: 10px;
  transition: all .2s;
}

.MenuItem figure:hover,
.MenuItem figure:focus,
.MenuItem figure:active {
  background-color: rgba(255,255,255,.65);
  box-shadow: 0 0 20px rgba(0,0,0,.2) inset;
}

.MenuItem figure strong {
  display: block;
}

.MenuItem .cheeseburger {
  background-position: 40px -165px;
}

.MenuItem .hotdog {
  background-position: -90px -165px;
}

.MenuItem .fries {
  background-position: -260px -172px;
}

.MenuItem .chicken {
  background-position: -412px -180px;
}

.MenuItem .coffee {
  background-position: 40px 24px;
}

.MenuItem .soda {
  background-position: -110px 28px;
}

.MenuItem .beer {
  background-position: -420px 30px;
}

.MenuItem .icecream {
  background-position: -415px -366px;
}
