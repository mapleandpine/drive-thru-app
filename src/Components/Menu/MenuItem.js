import React, { Component } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid';
import Menu from "./Menu.png";

class MenuItem extends Component {

  constructor(props) {
    super(props);
    this.state = {
      details:{
        id: uuid.v4(),
        class: this.props.menuitem.class,
        title: this.props.menuitem.title,
        price: this.props.menuitem.price
      }
    }
  }

  addItem(e) {
    this.props.onAddItem(this.state.details);
    e.preventDefault();
  }

  render() {
    var figureStyle = {
      backgroundImage: 'url(' + Menu + ')'
    };
    return (
      <div className="MenuItem" onClick={this.addItem.bind(this)}>
        <figure className={this.props.menuitem.class} style={figureStyle}>
          <strong>{this.props.menuitem.title}</strong> {this.props.menuitem.price}
        </figure>
      </div>
    );
  }
}

MenuItem.propTypes = {
  menuitem: PropTypes.object,
  onAddItem: PropTypes.func
}

export default MenuItem;
