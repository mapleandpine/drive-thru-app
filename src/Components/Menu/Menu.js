import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MenuItem from './MenuItem';
import "./Menu.css";

class Menu extends Component {

  addItem(id) {
    this.props.onAddItem(id);
  }

  render() {
    let menuItems;
    if(this.props.menu) {
      menuItems = this.props.menu.map(menuitem => {
        return (
          <MenuItem onAddItem={this.addItem.bind(this)} key={menuitem.class} menuitem={menuitem}/>
        )
      });
    }

    return (
      <div className="Menu">
        <div className="MenuList">
          {menuItems}
        </div>
      </div>
    );
  }
}

Menu.propTypes = {
  onAddItem: PropTypes.func,
  menu: PropTypes.object
}

export default Menu;

