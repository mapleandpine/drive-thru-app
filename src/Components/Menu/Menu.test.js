import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Menu from './Menu';

describe(Menu, () => {
  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <Menu />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
