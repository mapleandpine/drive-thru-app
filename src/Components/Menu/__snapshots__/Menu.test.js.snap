// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Menu extends _react.Component {

  addItem(id) {
    this.props.onAddItem(id);
  }

  render() {
    let menuItems;
    if (this.props.menu) {
      menuItems = this.props.menu.map(menuitem => {
        return (
          _react2.default.createElement(_MenuItem2.default, { onAddItem: this.addItem.bind(this), key: menuitem.class, menuitem: menuitem, __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }));

      });
    }

    return (
      _react2.default.createElement('div', { className: 'Menu', __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this },
        _react2.default.createElement('div', { className: 'MenuList', __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
          menuItems)));



  }} renders and matches our snapshot 1`] = `
<div
  className="Menu"
>
  <div
    className="MenuList"
  />
</div>
`;
